### DISCUSSION TITLE
Tweets' public_metrics not showing in response object with pagination

### DISCUSSION THREAD

**Discussion by 2jjk** [tweepy/tweepy#1808] [Questions] [answered]
Hi. Not sure if this is an issue.

I'm using `tweepy.Paginator` to get the most recent tweets in a set interval.

```
for response in tweepy.Paginator(client.search_recent_tweets, query=query, max_results=max_results, 
                start_time=start_time, end_time=end_time,
                tweet_fields=['created_at', 'public_metrics', 'lang', 'author_id'], expansions='author_id', 
                user_fields=['username', 'public_metrics', 'verified'], limit=5):
    print(response.data)
```

As I included the `tweet_fields `parameter with `public_metrics`, I expected to get the metrics in the data attribute of the Response object. But I only get the id of the tweet. Without pagination, it works. It also works when I called the `flatten` method on it with a limit (but then I can't use the `includes['users']` method on the object). I want the public metrics of both the tweet itself and the user posting it. 

Is that a limitation of the `search_recent_tweets` method or the sandbox environment I'm using? Or am I missing something here?

**Comment by Harmon758** [comment, marked as answer]
How are you attempting to access the metrics?

**Reply by 2jjk** [reply]
I tried` response.data['public_metrics']`,`response.data.public_metrics`, `response.public_metrics` and `response['public_metrics']`. I believe the results are truncated? Since printing the response object doesn't show any metrics. They should be in the data attribute of the response, correct?

**Reply by Harmon758** [reply]
No, they are [Tweet](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet) and [User](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/user) [fields](https://tweepy.readthedocs.io/en/v4.5.0/client.html#expansions-and-fields-parameters). You can access them as attributes or keys of the Tweet and User objects that are returned.

**Reply by 2jjk** [reply]
Sorry for the late response but that worked. I just looped over `response.data` and called tweet.public_metrics which returned a dictionary of what I need. A couple of things if I may; not sure if that's a Twitter API issue because I tried it with other python clients and the same results.

When I want to know the author of the tweet, I use `tweet.author_id` but that returns a random author_id of a user that retweeted the tweet. I'd have to exclude retweets.

And there doesn't seem to be a User object in the `response.data`. I want to get more info about the user but there's only the Tweet object, even with the expansions and user.fields.

**Reply by Harmon758** [reply]
> When I want to know the author of the tweet, I use `tweet.author_id` but that returns a random author_id of a user that retweeted the tweet. I'd have to exclude retweets.

You're getting the Retweet as a result. Retweets are Tweets, so the Retweeter is the author of the Retweet, not a random user.

> And there doesn't seem to be a User object in the `response.data`. I want to get more info about the user but there's only the Tweet object, even with the expansions and user.fields.

Expansion objects are returned in the `includes` field of the response.
https://tweepy.readthedocs.io/en/v4.5.0/client.html#expansions-and-fields-parameters
